### PR TITLE
Add a `dist-clean` and `dist-build` script to help with distribution.

### PR DESCRIPTION
**Pull request by demize**
Fairly minor PR just to help with distribution (also corrects the shebang in `build.ts` since I noticed that was wrong). 

@liquidnya, I'll send you the signing key I generated for this later, out-of-band.

### CODE REVIEW

File: dist.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+const version = process.env.npm_package_version;
+if (version == null) {
+  throw new Error("Package version undefined.");
+}
+
+const srcDir = dirname(fileURLToPath(import.meta.url));
+
+// Set up a temporary directory
+mkdtemp(join(tmpdir(), "fluidqueue-"), async (err, directory) => {
```
Comment (liquidnya) [inline dist.ts:20]: Is there a reason to use `fluidqueue-` instead of `fluid-queue-` (like everywhere else)?
(This doesn't really matter)

**Review by liquidnya — CHANGES_REQUESTED** [review]
- I wonder if we should also create a `.zip` version (for windows users).
- I think we should also put `CHANGELOG.md` into the distribution.
- I think we also should create a special `package.json` file, since we are distributing the `build` directory the `dependencies`, and `devDependencies` are maybe not needed? I wonder if the `optionalDependencies` are needed and how the build script handles those, so maybe for different systems a different build would need to be made?
- I can see how we are distributing the `LICENSE`, however we do not distribute the licenses for all dependencies even though we are distributing their sources (in a transpiled form). I wonder if we can export all licenses of all dependencies via `npm` or a different JavaScript tool and add that to the distribution.

**Comment by demize** [comment]
> I think we also should create a special package.json file, since we are distributing the build directory the dependencies, and devDependencies are maybe not needed?

We *could* but there's no real point to it. Though we will need to update the site to reflect the new setup method with packaged releases...

People don't need to run `npm install` at all with the packaged releases, since it's all bundled by esbuild; they can get started just by creating their settings files and doing `npm run start`. So while we could make a special package.json that doesn't include the dependencies (and it would probably be fine), there's no real point, since nobody should be installing them anyway. It would really just make the packaging process more complicated.

> I wonder if the optionalDependencies are needed and how the build script handles those, so maybe for different systems a different build would need to be made?

The optional dependencies are required for the build process; esbuild and swc both have platform-specific binaries they need to install. But we don't use esbuild or swc to *run* the bot; all the optional dependencies are children of dev dependencies. They won't be included in the bundled output, and we don't need to worry about them for distribution.

**Review by liquidnya — COMMENTED** [review]
The packaged archive now looks really good to me!

I wonder if we should also add the `oss-attribution` to our docker images, since we are also distributing an image that contains those dependencies (as transpiled sources).
Also when I was running `npm run dist-build` there was an error, because I didn't have `oss-attribution-generator` installed, and when I installed it with `npm i -g oss-attribution-generator` I got multiple warnings of deprecated libraries and also it has 7 vulnerabilities, and the latest release was 5 years ago.
Did you use a different tool that has the same name?
If not should we switch to a different tool?
I really like the outputted files of that tool though.

**Comment by demize** [comment]
No, that's the tool. Under different circumstances I'd make sure we were installing it to the project itself, but given a) it has 124+ dependencies, b) many of them are outdated, and c) it's very likely only I'll ever need to run this script, I felt this was the best solution.

There *are* other tools that can be used to parse the licenses of the project, but they don't actually produce the same output; this seems to be the best option for that. It may not have been updated in 5 years, but it works, and that's what really matters.

Would be good to include the attributions in the Docker container, though I'm not too concerned; they're not really accessible from the container unless you go digging for them, and at that point it's easier just to come check the repository directly. Docker is (unfortunately) pretty opaque to most users.

**Review by liquidnya — APPROVED** [review]
Alright, then we are good to merge then.

**Pull request merged**